snapshotter/volume_integ_test.go

```diff
@@ -0,0 +1,151 @@
+	"github.com/firecracker-microvm/firecracker-containerd/snapshotter/internal/integtest/stargz/fs/source"
+	"github.com/firecracker-microvm/firecracker-containerd/volume"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const mib = 1024 * 1024
+
+func TestGuestVolumeFrom_Isolated(t *testing.T) {
+	integtest.Prepare(t, integtest.WithDefaultNetwork())
+
+	runtimes := []string{
+		"aws.firecracker",
+		// "io.containerd.runc.v2"
+	}
+
+	for _, rt := range runtimes {
+		t.Run(rt, func(t *testing.T) {
+			testGuestVolumeFrom(t, rt)
+		})
+	}
+}
+
+func testGuestVolumeFrom(t *testing.T, runtime string) {
+	const vmID = "default"
+	al2 := imageRef
+	postgres := "docker.io/library/postgres:14.3"
+
+	ctx := namespaces.WithNamespace(context.Background(), vmID)
+
+	client, err := containerd.New(integtest.ContainerdSockPath, containerd.WithDefaultRuntime(runtime))
+	require.NoError(t, err, "unable to create client to containerd service at %s, is containerd running?", integtest.ContainerdSockPath)
+	defer client.Close()
+	fcClient, err := integtest.NewFCControlClient(integtest.ContainerdSockPath)
+	require.NoError(t, err, "Failed to create fccontrol client")
+
+	vs := volume.NewSet(runtime)
+	localImage := volume.FromImage(client, postgres, "ss1", volume.WithSnapshotter("devmapper"))
+	err = vs.AddFrom(ctx, localImage)
+	require.NoError(t, err)
+
+	remoteImage := volume.FromGuestImage(
+		client, vmID, imageRef, "ss2", []string{"/etc/yum"},
+		volume.WithSnapshotter("demux"),
+	)
+	mount, err := vs.PrepareDriveMount(ctx, 10*mib)
+	require.NoError(t, err)
+
+	_, err = fcClient.CreateVM(ctx, &proto.CreateVMRequest{
+		VMID: vmID,
+		RootDrive: &proto.FirecrackerRootDrive{
+			HostPath: "/var/lib/firecracker-containerd/runtime/rootfs-stargz.img",
+		},
+		NetworkInterfaces: []*proto.FirecrackerNetworkInterface{
+			{
+				AllowMMDS: true,
+				CNIConfig: &proto.CNIConfiguration{
+					NetworkName:   "fcnet",
+					InterfaceName: "veth0",
+				},
+			},
+		},
+		MachineCfg: &proto.FirecrackerMachineConfiguration{
+			VcpuCount:  2,
+			MemSizeMib: 2048,
+		},
+		ContainerCount: 1,
+		DriveMounts:    []*proto.FirecrackerDriveMount{mount},
+	})
+	require.NoErrorf(t, err, "Failed to create microVM[%s]", vmID)
+	defer fcClient.StopVM(ctx, &proto.StopVMRequest{VMID: vmID})
+
+	_, err = fcClient.SetVMMetadata(ctx, &proto.SetVMMetadataRequest{
+		VMID:     vmID,
+		Metadata: fmt.Sprintf(dockerMetadataTemplate, "ghcr.io", noAuth, noAuth),
+	})
+	require.NoError(t, err, "Failed to configure VM metadata for registry resolution")
+
+	err = remoteImage.Pull(ctx,
+		containerd.WithImageHandlerWrapper(source.AppendDefaultLabelsHandlerWrapper(imageRef, 10*mib)),
+	)
+	require.NoError(t, err)
+
+	err = remoteImage.Copy(ctx, "image")
+	require.NoError(t, err)
+
+	err = vs.AddFrom(ctx, remoteImage)
+	require.NoError(t, err)
+
+	image, err := client.Pull(ctx, "docker.io/library/alpine:3.10.1",
```

Comment:
There are a lot of hard-coded image references in this test, but also some declared in the beginning of the function as constants.  We should probably aim for more consistency in this regard.